config/samples/snapshot/snapshot.yaml

```diff
@@ -0,0 +1,14 @@
+# API type definition is in github.com/redhat-appstudio/managed-gitops
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ApplicationSnapshot
+metadata:
+  name: applicationsnapshot-sample
+spec:
+  application: application-sample
+  displayName: my-snapshot
+  displayDescription: my first snapshot
+  components:
```

Comment:
What if the same component is referenced more than once here? Is that valid / Do we check for that and validate?

Reply:
not sure how much more validation we want to do, I just did some basic sanity validation

I feel the gitops webhook should be validating these semantics or whoever is the primary controller. But your point stands, it is a valid case and i have not done it in this PR